src/main/java/net/earthcomputer/clientcommands/mixin/MixinItemGroup.java

```diff
@@ -0,0 +1,40 @@
+package net.earthcomputer.clientcommands.mixin;
+
+import net.earthcomputer.clientcommands.interfaces.IItemGroup;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(ItemGroup.class)
+public abstract class MixinItemGroup implements IItemGroup {
+
+    @Shadow @Mutable @Final public static ItemGroup[] GROUPS;
+
+    @Shadow @Mutable @Final private Text translationKey;
+
+    @Override
+    public void shrink(int index) {
+        ItemGroup[] tempGroups = GROUPS;
+        GROUPS = new ItemGroup[index];
+        System.arraycopy(tempGroups, 0, GROUPS, 0, GROUPS.length);
```

Comment:
This can be simplified to `Arrays.copyOf`. Also, was there a reason that `GROUPS` couldn't be accessed by an accessor mixin instead?

Reply:
`Arrays.copyOf` calls `System.arraycopy` internally, Are you sure you still want to use `Arrays.copyOf`?

As to the other question, I guess it could be 🤷‍♂️.

Reply:
I think I missed your response here, what was it again?

Reply:
The way it's been rewritten, you can just reassign `GROUPS` like I commented below.